File: frontend/src/services/cms.ts
import apiClient from '@/api/client';

const API_BASE_URL = '/v1';

export interface Category {
  id: string;
  name: string;
  code: string;
  description?: string;
  parent_id?: string;
  department_id?: string;
  sort_order: number;
  is_active: boolean;
  created_at: string;
  updated_at: string;
}

export interface Article {
  id: string;
  category_id: string;
  categoryId?: string;
  title: string;
  slug: string;
  summary?: string;
  content?: string;
  cover_image?: string;
  coverImage?: string;
  author_id: string;
  authorId?: string;
  department_id?: string;
  departmentId?: string;
  status: 'draft' | 'pending_review' | 'published' | 'rejected' | 'archived';
  view_count: number;
  viewCount?: number;
  is_top: boolean;
  isTop?: boolean;
  published_at?: string;
  publishedAt?: string;
  review_timeout_days?: number;
  reviewTimeoutDays?: number;
  review_reminded_at?: string;
  reviewRemindedAt?: string;
  is_overdue?: boolean;
  isOverdue?: boolean;
  remaining_hours?: number;
  remainingHours?: number;
  author_name?: string;
  authorName?: string;
  category_code?: string;
  categoryCode?: string;
  created_at: string;
  createdAt?: string;
  updated_at: string;
  updatedAt?: string;
}

export interface ArticleReview {
  id: string;
  article_id: string;
  reviewer_id: string;
  status: 'approved' | 'rejected' | 'requested_changes';
  comment?: string;
  reviewed_at?: string;
  created_at: string;
}

export interface CreateArticleRequest {
  category_code: string;
  title: string;
  summary?: string;
  content?: string;
  cover_image?: string;
  author_id?: string;
}

export interface UpdateArticleRequest {
  title?: string;
  summary?: string;
  content?: string;
  cover_image?: string;
}

export interface ReviewArticleRequest {
  status: 'approved' | 'rejected' | 'requested_changes';
  comment?: string;
}

export interface CreateCategoryRequest {
  name: string;
  code: string;
  description?: string;
  parent_id?: string;
  department_id?: string;
  sort_order?: number;
}

// Helper function to extract data from API response
const extractArrayData = (response: unknown): unknown[] => {
  const result = (response as Record<string, unknown>)?.data;
  const resultRecord = result as Record<string, unknown>;
  if (resultRecord?.success && resultRecord?.data) {
    if (Array.isArray(resultRecord.data)) {
      return resultRecord.data;
    }
    if (resultRecord.data && (resultRecord.data as Record<string, unknown>).data && Array.isArray((resultRecord.data as Record<string, unknown>).data)) {
      return (resultRecord.data as Record<string, unknown>).data as unknown[];
    }
  }
  if (resultRecord?.data && Array.isArray(resultRecord.data)) {
    return resultRecord.data;
  }
  if ((response as Record<string, unknown>)?.data && Array.isArray((response as Record<string, unknown>).data)) {
    return (response as Record<string, unknown>).data as unknown[];
  }
  return [];
};

// Helper function to extract single object from API response
const extractObjectData = (response: unknown): Record<string, unknown> => {
  const result = (response as Record<string, unknown>)?.data;
  const resultRecord = result as Record<string, unknown>;
  if (resultRecord?.success && resultRecord?.data) {
    return resultRecord.data as Record<string, unknown>;
  }
  if (resultRecord?.data) {
    return resultRecord.data as Record<string, unknown>;
  }
  if ((response as Record<string, unknown>)?.data) {
    return (response as Record<string, unknown>).data as Record<string, unknown>;
  }
  return {};
};

export const cmsApi = {
  // Categories - 调用真实的CMS分类API！
  getCategories: async (_departmentId?: string): Promise<Category[]> => {
    try {
      const params = _departmentId ? { department_id: _departmentId } : {};
      const response = await apiClient.get(`${API_BASE_URL}/cms/categories`, { params });
      return extractArrayData(response) as Category[];
    } catch (error) {
      console.error('获取文章分类失败:', error);
      // 提供后备数据
      return [
        { id: '1', name: '产品中心', code: 'products', sort_order: 1, is_active: true, created_at: '', updated_at: '' },
        { id: '2', name: '新闻动态', code: 'news', sort_order: 2, is_active: true, created_at: '', updated_at: '' },
        { id: '3', name: '关于我们', code: 'about', sort_order: 3, is_active: true, created_at: '', updated_at: '' },
        { id: '4', name: '服务项目', code: 'services', sort_order: 4, is_active: true, created_at: '', updated_at: '' },
        { id: '5', name: '案例展示', code: 'cases', sort_order: 5, is_active: true, created_at: '', updated_at: '' },
      ];
    }
  },

  getCategory: async (code: string): Promise<Category> => {
    try {
      const response = await apiClient.get(`${API_BASE_URL}/cms/categories/${code}`);
      return extractObjectData(response) as unknown as Category;
    } catch (error) {
      console.error('获取分类失败:', error);
      return {} as Category;
    }
  },

  createCategory: async (request: CreateCategoryRequest): Promise<Category> => {
    try {
      const response = await apiClient.post(`${API_BASE_URL}/cms/categories`, request);
      return extractObjectData(response) as unknown as Category;
    } catch (error) {
      console.error('创建分类失败:', error);
      return {} as Category;
    }
  },

  // Articles
  getArticles: async (status?: string, categoryCode?: string): Promise<Article[]> => {
    try {
      const params: Record<string, unknown> = {};
      if (status) params.status = status;
      if (categoryCode) params.category = categoryCode;
      const response = await apiClient.get(`${API_BASE_URL}/cms/articles`, { params });
      return extractArrayData(response) as Article[];
    } catch (error) {
      console.error('获取文章列表失败:', error);
      return [];
    }
  },

  getPublicArticles: async (categoryCode?: string): Promise<Article[]> => {
    try {
      const params: Record<string, unknown> = categoryCode ? { category: categoryCode } : {};
      const response = await apiClient.get(`${API_BASE_URL}/cms/public/articles`, { params });
      return extractArrayData(response) as Article[];
    } catch (error) {
      console.error('获取公开文章失败:', error);
      return [];
    }
  },

  getArticle: async (id: string): Promise<Article | null> => {
    try {
      const response = await apiClient.get(`${API_BASE_URL}/cms/articles/${id}`);
      return extractObjectData(response) as unknown as Article;
    } catch (error) {
      console.error('获取文章详情失败:', error);
      return null;
    }
  },

  getPublicArticle: async (id: string): Promise<Article | null> => {
    try {
      const response = await apiClient.get(`${API_BASE_URL}/cms/public/articles/${id}`);
      return extractObjectData(response) as unknown as Article;
    } catch (error) {
      console.error('获取公开文章详情失败:', error);
      return null;
    }
  },

  createArticle: async (request: CreateArticleRequest): Promise<Article> => {
    try {
      const response = await apiClient.post(`${API_BASE_URL}/cms/articles`, request);
      return extractObjectData(response) as unknown as Article;
    } catch (error) {
      console.error('创建文章失败:', error);
      throw error;
    }
  },

  updateArticle: async (id: string, request: UpdateArticleRequest): Promise<Article> => {
    try {
      const response = await apiClient.put(`${API_BASE_URL}/cms/articles/${id}`, request);
      return extractObjectData(response) as unknown as Article;
    } catch (error) {
      console.error('更新文章失败:', error);
      throw error;
    }
  },

  submitForReview: async (id: string): Promise<Article> => {
    try {
      const response = await apiClient.post(`${API_BASE_URL}/cms/articles/${id}/submit-review`);
      return extractObjectData(response) as unknown as Article;
    } catch (error) {
      console.error('提交审核失败:', error);
      throw error;
    }
  },

  reviewArticle: async (id: string, request: ReviewArticleRequest): Promise<ArticleReview> => {
    try {
      const response = await apiClient.post(`${API_BASE_URL}/cms/articles/${id}/review`, request);
      return extractObjectData(response) as unknown as ArticleReview;
    } catch (error) {
      console.error('审核文章失败:', error);
      throw error;
    }
  },

  getArticleReviews: async (id: string): Promise<ArticleReview[]> => {
    try {
      const response = await apiClient.get(`${API_BASE_URL}/cms/articles/${id}/reviews`);
      return extractArrayData(response) as ArticleReview[];
    } catch (error) {
      console.error('获取文章审核记录失败:', error);
      return [];
    }
  },

  getPendingReviewArticles: async (): Promise<Article[]> => {
    try {
      const response = await apiClient.get(`${API_BASE_URL}/cms/articles/pending-review`);
      return extractArrayData(response) as Article[];
    } catch (error) {
      console.error('获取待审核文章失败:', error);
      return [];
    }
  },

  getReviewedArticles: async (): Promise<Article[]> => {
    try {
      const response = await apiClient.get(`${API_BASE_URL}/cms/articles/reviewed`);
      return extractArrayData(response) as Article[];
    } catch (error) {
      console.error('获取已审核文章失败:', error);
      return [];
    }
  },
};

File: backend/core/src/api/handlers/cms.rs
use axum::{
    extract::{Path, Query, State},
    http::StatusCode,
    response::IntoResponse,
    Json,
};
use serde::{Deserialize, Serialize};
use std::sync::Arc;
use uuid::Uuid;

use crate::db::cms::{
    Article, ArticleReview, CreateArticle, CreateCategory, ReviewArticle, ReviewStats,
};
use crate::db::cms_repo::{
    archive_article, assign_reviewer, create_article, create_category, get_article,
    get_article_reviews, get_articles, get_categories, get_pending_review_articles,
    get_review_assignments, get_review_stats, get_reviewing_articles, get_reviewed_articles,
    publish_article, review_article, restore_article_from_archive, submit_article_for_review,
    update_article,
};
use crate::errors::ApiResponse;
use crate::state::AppState;

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CategoryQueryParams {
    pub department_id: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ArticleQueryParams {
    pub status: Option<String>,
    pub category: Option<String>,
    pub page: Option<i32>,
    pub page_size: Option<i32>,
}

pub async fn get_categories_handler(
    State(state): State<Arc<AppState>>,
    Query(params): Query<CategoryQueryParams>,
) -> impl IntoResponse {
    let department_id = params
        .department_id
        .and_then(|id| Uuid::parse_str(&id).ok());

    match get_categories(&state.db, department_id).await {
        Ok(categories) => (StatusCode::OK, Json(ApiResponse::success(categories))),
        Err(e) => (
            StatusCode::INTERNAL_SERVER_ERROR,
            Json(ApiResponse::<Vec<crate::db::cms::Category>>::error(
                e.to_string(),
            )),
        ),
    }
}

pub async fn create_category_handler(
    State(state): State<Arc<AppState>>,
    Json(req): Json<CreateCategoryRequest>,
) -> impl IntoResponse {
    let create_req = CreateCategory {
        name: req.name,
        code: req.code,
        description: req.description,
        parent_id: req.parent_id.and_then(|id| Uuid::parse_str(&id).ok()),
        department_id: req.department_id.and_then(|id| Uuid::parse_str(&id).ok()),
        sort_order: req.sort_order,
    };

    match create_category(&state.db, create_req).await {
        Ok(category) => (StatusCode::CREATED, Json(ApiResponse::success(category))),
        Err(e) => (
            StatusCode::INTERNAL_SERVER_ERROR,
            Json(ApiResponse::<crate::db::cms::Category>::error(
                e.to_string(),
            )),
        ),
    }
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CreateCategoryRequest {
    pub name: String,
    pub code: String,
    pub description: Option<String>,
    pub parent_id: Option<String>,
    pub department_id: Option<String>,
    pub sort_order: Option<i32>,
}

pub async fn get_articles_handler(
    State(state): State<Arc<AppState>>,
    Query(params): Query<ArticleQueryParams>,
) -> impl IntoResponse {
    let search_params = crate::db::cms::ArticleSearchParams {
        status: params.status,
        category_code: params.category,
        page: params.page,
        page_size: params.page_size,
    };

    match get_articles(&state.db, search_params).await {
        Ok(articles) => (StatusCode::OK, Json(ApiResponse::success(articles))),
        Err(e) => (
            StatusCode::INTERNAL_SERVER_ERROR,
            Json(ApiResponse::<Vec<Article>>::error(e.to_string())),
        ),
    }
}

pub async fn get_article_handler(
    State(state): State<Arc<AppState>>,
    Path(id): Path<String>,
) -> impl IntoResponse {
    let article_id = match Uuid::parse_str(&id) {
        Ok(uuid) => uuid,
        Err(_) => {
            return (
                StatusCode::BAD_REQUEST,
                Json(ApiResponse::<Article>::error(
                    "Invalid article ID".to_string(),
                )),
            )
        }
    };

    match get_article(&state.db, article_id).await {
        Ok(Some(article)) => (StatusCode::OK, Json(ApiResponse::success(article))),
        Ok(None) => (
            StatusCode::NOT_FOUND,
            Json(ApiResponse::<Article>::error(
                "Article not found".to_string(),
            )),
        ),
        Err(e) => (
            StatusCode::INTERNAL_SERVER_ERROR,
            Json(ApiResponse::<Article>::error(e.to_string())),
        ),
    }
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CreateArticleRequest {
    pub category_code: String,
    pub title: String,
    pub summary: Option<String>,
    pub content: Option<String>,
    pub cover_image: Option<String>,
    pub author_id: Option<String>,
}

pub async fn create_article_handler(
    State(state): State<Arc<AppState>>,
    Json(req): Json<CreateArticleRequest>,
) -> impl IntoResponse {
    let author_id = req
        .author_id
        .and_then(|id| Uuid::parse_str(&id).ok())
        .unwrap_or_else(|| Uuid::parse_str("7f6d2c08-7e18-41ea-81f1-7b1682995352").unwrap());

    let create_req = CreateArticle {
        category_code: req.category_code,
        title: req.title,
        summary: req.summary,
        content: req.content,
        cover_image: req.cover_image,
        author_id,
    };

    match create_article(&state.db, create_req).await {
        Ok(article) => (StatusCode::CREATED, Json(ApiResponse::success(article))),
        Err(e) => (
            StatusCode::INTERNAL_SERVER_ERROR,
            Json(ApiResponse::<Article>::error(e.to_string())),
        ),
    }
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct UpdateArticleRequest {
    pub title: Option<String>,
    pub summary: Option<String>,
    pub content: Option<String>,
    pub cover_image: Option<String>,
}

pub async fn update_article_handler(
    State(state): State<Arc<AppState>>,
    Path(id): Path<String>,
    Json(req): Json<UpdateArticleRequest>,
) -> impl IntoResponse {
    let article_id = match Uuid::parse_str(&id) {
        Ok(uuid) => uuid,
        Err(_) => {
            return (
                StatusCode::BAD_REQUEST,
                Json(ApiResponse::<Article>::error(
                    "Invalid article ID".to_string(),
                )),
            )
        }
    };

    let update_req = crate::db::cms::UpdateArticle {
        title: req.title,
        summary: req.summary,
        content: req.content,
        cover_image: req.cover_image,
    };

    match update_article(&state.db, article_id, update_req).await {
        Ok(Some(article)) => (StatusCode::OK, Json(ApiResponse::success(article))),
        Ok(None) => (
            StatusCode::NOT_FOUND,
            Json(ApiResponse::<Article>::error(
                "Article not found or cannot be updated".to_string(),
            )),
        ),
        Err(e) => (
            StatusCode::INTERNAL_SERVER_ERROR,
            Json(ApiResponse::<Article>::error(e.to_string())),
        ),
    }
}

pub async fn submit_article_for_review_handler(
    State(state): State<Arc<AppState>>,
    Path(id): Path<String>,
) -> impl IntoResponse {
    let article_id = match Uuid::parse_str(&id) {
        Ok(uuid) => uuid,
        Err(_) => {
            return (
                StatusCode::BAD_REQUEST,
                Json(ApiResponse::<Article>::error(
                    "Invalid article ID".to_string(),
                )),
            )
        }
    };

    let user_id = Uuid::parse_str("7f6d2c08-7e18-41ea-81f1-7b1682995352").unwrap();

    match submit_article_for_review(&state.db, article_id, user_id).await {
        Ok(Some(article)) => (StatusCode::OK, Json(ApiResponse::success(article))),
        Ok(None) => (
            StatusCode::NOT_FOUND,
            Json(ApiResponse::<Article>::error(
                "Article not found".to_string(),
            )),
        ),
        Err(e) => (
            StatusCode::INTERNAL_SERVER_ERROR,
            Json(ApiResponse::<Article>::error(e.to_string())),
        ),
    }
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct AssignReviewerRequest {
    pub reviewer_id: String,
}

pub async fn assign_reviewer_handler(
    State(state): State<Arc<AppState>>,
    Path(id): Path<String>,
    Json(req): Json<AssignReviewerRequest>,
) -> impl IntoResponse {
    let article_id = match Uuid::parse_str(&id) {
        Ok(uuid) => uuid,
        Err(_) => {
            return (
                StatusCode::BAD_REQUEST,
                Json(ApiResponse::<Article>::error(
                    "Invalid article ID".to_string(),
                )),
            )
        }
    };

    let reviewer_id = match Uuid::parse_str(&req.reviewer_id) {
        Ok(uuid) => uuid,
        Err(_) => {
            return (
                StatusCode::BAD_REQUEST,
                Json(ApiResponse::<Article>::error(
                    "Invalid reviewer ID".to_string(),
                )),
            )
        }
    };

    match assign_reviewer(&state.db, article_id, reviewer_id).await {
        Ok(Some(article)) => (StatusCode::OK, Json(ApiResponse::success(article))),
        Ok(None) => (
            StatusCode::NOT_FOUND,
            Json(ApiResponse::<Article>::error(
                "Article not found or not pending review".to_string(),
            )),
        ),
        Err(e) => (
            StatusCode::INTERNAL_SERVER_ERROR,
            Json(ApiResponse::<Article>::error(e.to_string())),
        ),
    }
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct ReviewArticleRequest {
    pub status: String,
    pub comment: Option<String>,
}

pub async fn review_article_handler(
    State(state): State<Arc<AppState>>,
    Path(id): Path<String>,
    Json(req): Json<ReviewArticleRequest>,
) -> impl IntoResponse {
    let article_id = match Uuid::parse_str(&id) {
        Ok(uuid) => uuid,
        Err(_) => {
            return (
                StatusCode::BAD_REQUEST,
                Json(ApiResponse::<ArticleReview>::error(
                    "Invalid article ID".to_string(),
                )),
            )
        }
    };

    let reviewer_id = Uuid::parse_str("7f6d2c08-7e18-41ea-81f1-7b1682995352").unwrap();

    let review_req = ReviewArticle {
        status: req.status,
        comment: req.comment,
    };

    match review_article(&state.db, article_id, reviewer_id, review_req).await {
        Ok(Some(review)) => (StatusCode::OK, Json(ApiResponse::success(review))),
        Ok(None) => (
            StatusCode::NOT_FOUND,
            Json(ApiResponse::<ArticleReview>::error(
                "Article not found".to_string(),
            )),
        ),
        Err(e) => (
            StatusCode::INTERNAL_SERVER_ERROR,
            Json(ApiResponse::<ArticleReview>::error(e.to_string())),
        ),
    }
}

pub async fn publish_article_handler(
    State(state): State<Arc<AppState>>,
    Path(id): Path<String>,
) -> impl IntoResponse {
    let article_id = match Uuid::parse_str(&id) {
        Ok(uuid) => uuid,
        Err(_) => {
            return (
                StatusCode::BAD_REQUEST,
                Json(ApiResponse::<Article>::error(
                    "Invalid article ID".to_string(),
                )),
            )
        }
    };

    match publish_article(&state.db, article_id).await {
        Ok(Some(article)) => (StatusCode::OK, Json(ApiResponse::success(article))),
        Ok(None) => (
            StatusCode::NOT_FOUND,
            Json(ApiResponse::<Article>::error(
                "Article not found or not approved".to_string(),
            )),
        ),
        Err(e) => (
            StatusCode::INTERNAL_SERVER_ERROR,
            Json(ApiResponse::<Article>::error(e.to_string())),
        ),
    }
}

pub async fn archive_article_handler(
    State(state): State<Arc<AppState>>,
    Path(id): Path<String>,
) -> impl IntoResponse {
    let article_id = match Uuid::parse_str(&id) {
        Ok(uuid) => uuid,
        Err(_) => {
            return (
                StatusCode::BAD_REQUEST,
                Json(ApiResponse::<Article>::error(
                    "Invalid article ID".to_string(),
                )),
            )
        }
    };

    match archive_article(&state.db, article_id).await {
        Ok(Some(article)) => (StatusCode::OK, Json(ApiResponse::success(article))),
        Ok(None) => (
            StatusCode::NOT_FOUND,
            Json(ApiResponse::<Article>::error(
                "Article not found".to_string(),
            )),
        ),
        Err(e) => (
            StatusCode::INTERNAL_SERVER_ERROR,
            Json(ApiResponse::<Article>::error(e.to_string())),
        ),
    }
}

pub async fn restore_article_handler(
    State(state): State<Arc<AppState>>,
    Path(id): Path<String>,
) -> impl IntoResponse {
    let article_id = match Uuid::parse_str(&id) {
        Ok(uuid) => uuid,
        Err(_) => {
            return (
                StatusCode::BAD_REQUEST,
                Json(ApiResponse::<Article>::error(
                    "Invalid article ID".to_string(),
                )),
            )
        }
    };

    match restore_article_from_archive(&state.db, article_id).await {
        Ok(Some(article)) => (StatusCode::OK, Json(ApiResponse::success(article))),
        Ok(None) => (
            StatusCode::NOT_FOUND,
            Json(ApiResponse::<Article>::error(
                "Article not found or not archived".to_string(),
            )),
        ),
        Err(e) => (
            StatusCode::INTERNAL_SERVER_ERROR,
            Json(ApiResponse::<Article>::error(e.to_string())),
        ),
    }
}

pub async fn get_article_reviews_handler(
    State(state): State<Arc<AppState>>,
    Path(id): Path<String>,
) -> impl IntoResponse {
    let article_id = match Uuid::parse_str(&id) {
        Ok(uuid) => uuid,
        Err(_) => {
            return (
                StatusCode::BAD_REQUEST,
                Json(ApiResponse::<Vec<ArticleReview>>::error(
                    "Invalid article ID".to_string(),
                )),
            )
        }
    };

    match get_article_reviews(&state.db, article_id).await {
        Ok(reviews) => (StatusCode::OK, Json(ApiResponse::success(reviews))),
        Err(e) => (
            StatusCode::INTERNAL_SERVER_ERROR,
            Json(ApiResponse::<Vec<ArticleReview>>::error(e.to_string())),
        ),
    }
}

pub async fn get_pending_review_articles_handler(
    State(state): State<Arc<AppState>>,
) -> impl IntoResponse {
    match get_pending_review_articles(&state.db).await {
        Ok(articles) => (StatusCode::OK, Json(ApiResponse::success(articles))),
        Err(e) => (
            StatusCode::INTERNAL_SERVER_ERROR,
            Json(ApiResponse::<Vec<Article>>::error(e.to_string())),
        ),
    }
}

pub async fn get_reviewing_articles_handler(
    State(state): State<Arc<AppState>>,
) -> impl IntoResponse {
    match get_reviewing_articles(&state.db, None).await {
        Ok(articles) => (StatusCode::OK, Json(ApiResponse::success(articles))),
        Err(e) => (
            StatusCode::INTERNAL_SERVER_ERROR,
            Json(ApiResponse::<Vec<Article>>::error(e.to_string())),
        ),
    }
}

pub async fn get_reviewed_articles_handler(
    State(state): State<Arc<AppState>>,
) -> impl IntoResponse {
    match get_reviewed_articles(&state.db).await {
        Ok(articles) => (StatusCode::OK, Json(ApiResponse::success(articles))),
        Err(e) => (
            StatusCode::INTERNAL_SERVER_ERROR,
            Json(ApiResponse::<Vec<Article>>::error(e.to_string())),
        ),
    }
}

pub async fn get_review_stats_handler(
    State(state): State<Arc<AppState>>,
) -> impl IntoResponse {
    match get_review_stats(&state.db).await {
        Ok(stats) => (StatusCode::OK, Json(ApiResponse::success(stats))),
        Err(e) => (
            StatusCode::INTERNAL_SERVER_ERROR,
            Json(ApiResponse::<ReviewStats>::error(e.to_string())),
        ),
    }
}

pub async fn get_review_assignments_handler(
    State(state): State<Arc<AppState>>,
) -> impl IntoResponse {
    let reviewer_id = Uuid::parse_str("7f6d2c08-7e18-41ea-81f1-7b1682995352").unwrap();

    match get_review_assignments(&state.db, reviewer_id).await {
        Ok(assignments) => (StatusCode::OK, Json(ApiResponse::success(assignments))),
        Err(e) => (
            StatusCode::INTERNAL_SERVER_ERROR,
            Json(ApiResponse::<Vec<crate::db::cms::ReviewAssignment>>::error(e.to_string())),
        ),
    }
}

pub async fn get_public_articles_handler(
    State(state): State<Arc<AppState>>,
    Query(params): Query<ArticleQueryParams>,
) -> impl IntoResponse {
    let search_params = crate::db::cms::ArticleSearchParams {
        status: Some("published".to_string()),
        category_code: params.category,
        page: params.page,
        page_size: params.page_size,
    };

    match get_articles(&state.db, search_params).await {
        Ok(articles) => (StatusCode::OK, Json(ApiResponse::success(articles))),
        Err(e) => (
            StatusCode::INTERNAL_SERVER_ERROR,
            Json(ApiResponse::<Vec<Article>>::error(e.to_string())),
        ),
    }
}